python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -599,6 +605,186 @@ async def verify_calib(self, image_type, job_id_calib, exposure_ids_dict):
+            )
+        butler = dafButler.Butler(
+            self.config.repo, collections=[verifyCollection, genCollection]
+        )
+        # verify_stats is a dictionary with the verification
+        # tests that failed, if any. See `cp_verify`.
+        verify_stats = butler.get(verify_stats_string, instrument=self.instrument_name)
+
+        if "FAILURES" in verify_stats:
+            (
+                certify_calib,
+                numStatErrors,
+                failure_thresholds,
+            ) = await self.count_failed_verification_tests(verify_stats)
+
+            return certify_calib, numStatErrors, failure_thresholds, verify_stats
+        else:
+            # Nothing failed
+            return True, None, None, verify_stats
+
+    async def count_failed_verification_tests(self, verify_stats):
+        """Count number of tests that failed cp_verify.
+
+        Parameters
+        ----------
+        verify_stats : `dict`
+            Statistics from cp_verify.
+
+        Returns
+        -------
+        certify_calib : `bool`
+            Boolean assessing whether the calibration should be certified.
+
+        total_counter_failed_tests : `dict`[`str`][`str`]
+            Dictionary with the total number of tests failed per exposure and
+            per cp_verify test type. If there are not any tests that failed,
+            `None` will be returned.
+
+        thresholds : `dict`[`str`][`int`]
+            Dictionary reporting the different thresholds used to decide
+            whether a calibration shoudl be certified or not (see `Notes`
+            below).
+
+        Notes
+        -----
+        For at least one type of test, if the majority of tests fail in
+        the majority of detectors and the majority of exposures,
+        then don't certify the calibration.
+        """
+        certify_calib = True
+
+        # Thresholds
+
+        min_number_failures_per_detector_per_test = (
+            self.config.number_verification_tests_threshold
+        )
+        # Main key of verify_stats is exposure IDs
+        min_number_failed_exposures = int(len(verify_stats) / 2) + 1  # majority of exps
+
+        first_exp = list(verify_stats.keys())[0]
+        # "stg" is of the form "detector_amp_test", and "detector" is
+        # of the form "raft_det".
+        detectors = set(
+            [stg.split(" ")[0] for stg in verify_stats[first_exp]["FAILURES"]]
+        )
+        min_number_failed_detectors = (
+            int(len(detectors) / 2) + 1
+        )  # majority of detectors
+
+        # Define failure threshold per exposure
+        failure_threshold_exposure = (
+            min_number_failures_per_detector_per_test * min_number_failed_detectors
+        )
+
+        # Count the number of failures per test per exposure.
+        total_counter_failed_tests = {}
+        # Pop this key so that we are left with
+        # exposure ID's as keys only.
+        verify_stats.pop("SUCCESS")
```

Comment:
This line is changing the input dictionary, are you sure you want to do that? You may want to create a local copy of it to operate if you intend on doing this. But you could also do something like:

```
for exposure in [key for key in verify_stats if key != "SUCCESS"]:
```